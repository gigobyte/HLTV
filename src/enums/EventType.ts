export enum EventType {
  Online = 'Online',
  LocalLan = 'Local LAN',
  IntlLan = 'International LAN',
  RegLan = 'Regional LAN',
  Major = 'Major',
  Other = 'Other'
}
