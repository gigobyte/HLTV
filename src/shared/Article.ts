export interface Article {
  name: string
  link: string
}
