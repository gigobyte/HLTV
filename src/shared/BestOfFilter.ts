export enum BestOfFilter {
  'BO1' = 'BestOf1',
  'BO3' = 'BestOf3',
  'BO5' = 'BestOf5'
}
