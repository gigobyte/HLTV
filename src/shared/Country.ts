export interface Country {
  name: string
  code: string
}
