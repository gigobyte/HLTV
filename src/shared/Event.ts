export interface Event {
  name: string
  id?: number
}
