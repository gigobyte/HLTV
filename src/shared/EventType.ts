export enum EventType {
  Major = 'MAJOR',
  InternationalLAN = 'INTLLAN',
  RegionalLAN = 'REGIONALLAN',
  LocalLAN = 'LOCALLAN',
  Online = 'ONLINE',
  Other = 'OTHER'
}
