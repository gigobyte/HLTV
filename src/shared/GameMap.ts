export enum GameMap {
  TBA = 'tba',
  Train = 'de_train',
  Cobblestone = 'de_cbble',
  Inferno = 'de_inferno',
  Cache = 'de_cache',
  Mirage = 'de_mirage',
  Overpass = 'de_overpass',
  Dust2 = 'de_dust2',
  Nuke = 'de_nuke',
  Tuscan = 'de_tuscan',
  Vertigo = 'de_vertigo',
  Season = 'de_season',
  Ancient = 'de_ancient',
  Anubis = 'de_anubis',
  Default = 'default'
}

export const fromMapSlug = (slug: string): GameMap => {
  switch (slug) {
    case 'tba':
      return GameMap.TBA
    case 'trn':
      return GameMap.Train
    case 'cbl':
      return GameMap.Cobblestone
    case 'inf':
      return GameMap.Inferno
    case 'cch':
      return GameMap.Cache
    case 'mrg':
      return GameMap.Mirage
    case 'ovp':
      return GameMap.Overpass
    case 'd2':
      return GameMap.Dust2
    case 'nuke':
      return GameMap.Nuke
    case 'tcn':
      return GameMap.Tuscan
    case 'vtg':
      return GameMap.Vertigo
    case 'anc':
      return GameMap.Ancient
    case '-':
      return GameMap.Default
    default:
      return GameMap.Default
  }
}

export const fromMapName = (name: string): GameMap => {
  switch (name) {
    case 'TBA':
      return GameMap.TBA
    case 'Train':
    case 'Train_se':
      return GameMap.Train
    case 'Cobblestone':
      return GameMap.Cobblestone
    case 'Inferno':
    case 'Inferno_se':
      return GameMap.Inferno
    case 'Cache':
      return GameMap.Cache
    case 'Mirage':
    case 'Mirage_ce':
      return GameMap.Mirage
    case 'Overpass':
      return GameMap.Overpass
    case 'Dust2':
    case 'Dust2_se':
      return GameMap.Dust2
    case 'Nuke':
    case 'Nuke_se':
      return GameMap.Nuke
    case 'Tuscan':
      return GameMap.Tuscan
    case 'Vertigo':
      return GameMap.Vertigo
    case 'Ancient':
      return GameMap.Ancient
    case 'Anubis':
      return GameMap.Anubis
    case 'Default':
      return GameMap.Default
    default:
      return GameMap.Default
  }
}

export const toMapFilter = (map: GameMap): string => {
  switch (map) {
    case GameMap.Cache:
      return 'de_cache'
    case GameMap.Cobblestone:
      return 'de_cobblestone'
    case GameMap.Overpass:
      return 'de_overpass'
    case GameMap.Dust2:
      return 'de_dust2'
    case GameMap.Inferno:
      return 'de_inferno'
    case GameMap.Mirage:
      return 'de_mirage'
    case GameMap.Nuke:
      return 'de_nuke'
    case GameMap.Train:
      return 'de_train'
    case GameMap.Tuscan:
      return 'de_tuscan'
    case GameMap.Vertigo:
      return 'de_vertigo'
    case GameMap.Ancient:
      return 'de_ancient'
    case GameMap.Anubis:
      return 'de_anubis'
    case GameMap.Season:
      return 'de_season'
    case GameMap.TBA:
    case GameMap.Default: {
      throw new Error(`Invalid map filter - ${map}`)
    }
  }
}
