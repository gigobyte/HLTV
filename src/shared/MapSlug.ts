export enum MapSlug {
  TBA = 'tba',
  Train = 'trn',
  Cobblestone = 'cbl',
  Inferno = 'inf',
  Cache = 'cch',
  Mirage = 'mrg',
  Overpass = 'ovp',
  Dust2 = 'd2',
  Nuke = 'nuke',
  Tuscan = 'tcn',
  Vertigo = 'vertigo',
  Season = '-',
  Default = '-'
}

export const toMapSlug = (map: string): MapSlug => (MapSlug as any)[map]
