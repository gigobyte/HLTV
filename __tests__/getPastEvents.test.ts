import HLTV from '../src'
import { EventType } from '../src/shared/EventType'

test('getPastEvents', async () => {
  expect(
    await HLTV.getPastEvents({
      startDate: '2017-01-01',
      endDate: '2017-12-31',
      eventType: EventType.InternationalLAN,
      delayBetweenPageRequests: 1500
    })
  ).toMatchSnapshot()
}, 30000)
