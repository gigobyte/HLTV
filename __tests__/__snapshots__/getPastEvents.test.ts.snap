// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getPastEvents 1`] = `
Array [
  Object {
    "dateEnd": 1514113200000,
    "dateStart": 1513854000000,
    "id": 3464,
    "location": Object {
      "code": "RS",
      "name": "Belgrade, Serbia",
    },
    "name": "Winner of the Future 2017",
    "numberOfTeams": 8,
    "prizePool": "$4,725",
  },
  Object {
    "dateEnd": 1513508400000,
    "dateStart": 1513335600000,
    "id": 3313,
    "location": Object {
      "code": "MX",
      "name": "Cancun, Mexico",
    },
    "name": "ECS Season 4 Finals",
    "numberOfTeams": 8,
    "prizePool": "$660,000",
  },
  Object {
    "dateEnd": 1512903600000,
    "dateStart": 1512471600000,
    "id": 3072,
    "location": Object {
      "code": "DK",
      "name": "Odense, Denmark",
    },
    "name": "ESL Pro League Season 6 Finals",
    "numberOfTeams": 12,
    "prizePool": "$750,000",
  },
  Object {
    "dateEnd": 1512817200000,
    "dateStart": 1512558000000,
    "id": 3283,
    "location": Object {
      "code": "MY",
      "name": "Kuala Lumpur, Malaysia",
    },
    "name": "ROG MASTERS 2017 Grand Finals",
    "numberOfTeams": 6,
    "prizePool": "$235,000",
  },
  Object {
    "dateEnd": 1512817200000,
    "dateStart": 1507370400000,
    "id": 3201,
    "location": Object {
      "code": "GB",
      "name": "London, United Kingdom",
    },
    "name": "Gfinity Elite Series Season 2",
    "numberOfTeams": 8,
    "prizePool": "$110,000",
  },
  Object {
    "dateEnd": 1512298800000,
    "dateStart": 1512126000000,
    "id": 2574,
    "location": Object {
      "code": "SE",
      "name": "Jönköping, Sweden",
    },
    "name": "DreamHack Open Winter 2017",
    "numberOfTeams": 8,
    "prizePool": "$100,000",
  },
  Object {
    "dateEnd": 1511694000000,
    "dateStart": 1511607600000,
    "id": 3224,
    "location": Object {
      "code": "BY",
      "name": "Minsk, Belarus",
    },
    "name": "FCDB Cup 2017",
    "numberOfTeams": 8,
    "prizePool": "$40,000",
  },
  Object {
    "dateEnd": 1511694000000,
    "dateStart": 1511434800000,
    "id": 3115,
    "location": Object {
      "code": "ES",
      "name": "Barcelona, Spain",
    },
    "name": "WESG 2017 Europe & CIS Regional Finals Female",
    "numberOfTeams": 8,
    "prizePool": "$16,000",
  },
  Object {
    "dateEnd": 1511694000000,
    "dateStart": 1511434800000,
    "id": 3114,
    "location": Object {
      "code": "ES",
      "name": "Barcelona, Spain",
    },
    "name": "WESG 2017 Europe & CIS Regional Finals",
    "numberOfTeams": 20,
    "prizePool": "$92,500",
  },
  Object {
    "dateEnd": 1511607600000,
    "dateStart": 1511521200000,
    "id": 2998,
    "location": Object {
      "code": "DK",
      "name": "Copenhagen, Denmark",
    },
    "name": "BLAST Pro Series Copenhagen 2017",
    "numberOfTeams": 6,
    "prizePool": "$250,000",
  },
  Object {
    "dateEnd": 1511434800000,
    "dateStart": 1511089200000,
    "id": 3292,
    "location": Object {
      "code": "CN",
      "name": "Shenzhen, China",
    },
    "name": "China Top 2017 Shenzhen",
    "numberOfTeams": 4,
    "prizePool": "$113,361",
  },
  Object {
    "dateEnd": 1511089200000,
    "dateStart": 1510657200000,
    "id": 2956,
    "location": Object {
      "code": "US",
      "name": "Burbank & Oakland, California, USA",
    },
    "name": "IEM Oakland 2017",
    "numberOfTeams": 12,
    "prizePool": "$300,000",
  },
  Object {
    "dateEnd": 1511002800000,
    "dateStart": 1510916400000,
    "id": 3355,
    "location": Object {
      "code": "KR",
      "name": "Busan, South Korea",
    },
    "name": "WEGL 2017 Super Fight Invitational",
    "numberOfTeams": 4,
    "prizePool": "$53,580",
  },
  Object {
    "dateEnd": 1511002800000,
    "dateStart": 1510916400000,
    "id": 3166,
    "location": Object {
      "code": "AU",
      "name": "Sydney, Australia",
    },
    "name": "ZEN League Season 2 Finals",
    "numberOfTeams": 4,
    "prizePool": "$24,300",
  },
  Object {
    "dateEnd": 1510484400000,
    "dateStart": 1510398000000,
    "id": 3165,
    "location": Object {
      "code": "US",
      "name": "Santa Ana, California, USA",
    },
    "name": "iBUYPOWER Masters 2017",
    "numberOfTeams": 8,
    "prizePool": "$100,000",
  },
  Object {
    "dateEnd": 1510398000000,
    "dateStart": 1510138800000,
    "id": 3069,
    "location": Object {
      "code": "KR",
      "name": "Busan, South Korea",
    },
    "name": "IeSF World Championship 2017",
    "numberOfTeams": 16,
    "prizePool": "$17,978",
  },
  Object {
    "dateEnd": 1509879600000,
    "dateStart": 1509620400000,
    "id": 3252,
    "location": Object {
      "code": "CA",
      "name": "Toronto, Canada",
    },
    "name": "Americas Minor - ELEAGUE Major 2018",
    "numberOfTeams": 8,
    "prizePool": "Other",
  },
  Object {
    "dateEnd": 1509879600000,
    "dateStart": 1509620400000,
    "id": 3251,
    "location": Object {
      "code": "RO",
      "name": "Bucharest, Romania",
    },
    "name": "Europe Minor - ELEAGUE Major 2018",
    "numberOfTeams": 8,
    "prizePool": "Other",
  },
  Object {
    "dateEnd": 1509879600000,
    "dateStart": 1509620400000,
    "id": 3190,
    "location": Object {
      "code": "CN",
      "name": "Shanghai, China",
    },
    "name": "SL i-League Invitational Shanghai 2017",
    "numberOfTeams": 8,
    "prizePool": "$150,000",
  },
  Object {
    "dateEnd": 1509271200000,
    "dateStart": 1509012000000,
    "id": 3250,
    "location": Object {
      "code": "RO",
      "name": "Bucharest, Romania",
    },
    "name": "CIS Minor - ELEAGUE Major 2018",
    "numberOfTeams": 8,
    "prizePool": "Other",
  },
  Object {
    "dateEnd": 1509271200000,
    "dateStart": 1509012000000,
    "id": 3249,
    "location": Object {
      "code": "KR",
      "name": "Seoul, South Korea",
    },
    "name": "Asia Minor - ELEAGUE Major 2018",
    "numberOfTeams": 8,
    "prizePool": "Other",
  },
  Object {
    "dateEnd": 1509271200000,
    "dateStart": 1508839200000,
    "id": 3133,
    "location": Object {
      "code": "RU",
      "name": "Saint Petersburg, Russia",
    },
    "name": "EPICENTER 2017",
    "numberOfTeams": 8,
    "prizePool": "$490,000",
  },
  Object {
    "dateEnd": 1509184800000,
    "dateStart": 1509098400000,
    "id": 3204,
    "location": Object {
      "code": "ES",
      "name": "Madrid, Spain",
    },
    "name": "ESL Clash of Nations",
    "numberOfTeams": 8,
    "prizePool": "$10,000",
  },
  Object {
    "dateEnd": 1508752800000,
    "dateStart": 1508752800000,
    "id": 3153,
    "location": Object {
      "code": "RU",
      "name": "Saint Petersburg, Russia",
    },
    "name": "EPICENTER 2017 Wild Card Qualifier",
    "numberOfTeams": 4,
    "prizePool": "Other",
  },
  Object {
    "dateEnd": 1508666400000,
    "dateStart": 1508493600000,
    "id": 2573,
    "location": Object {
      "code": "US",
      "name": "Denver, Colorado, USA",
    },
    "name": "DreamHack Open Denver 2017",
    "numberOfTeams": 8,
    "prizePool": "$100,000",
  },
  Object {
    "dateEnd": 1508666400000,
    "dateStart": 1508407200000,
    "id": 2921,
    "location": Object {
      "code": "CN",
      "name": "Shanghai, China",
    },
    "name": "eXTREMESLAND 2017 Asia Finals",
    "numberOfTeams": 16,
    "prizePool": "$100,000",
  },
  Object {
    "dateEnd": 1508061600000,
    "dateStart": 1507975200000,
    "id": 3183,
    "location": Object {
      "code": "DE",
      "name": "Krefeld, Germany",
    },
    "name": "ROG MASTERS 2017 EMEA Finals",
    "numberOfTeams": 8,
    "prizePool": "Other",
  },
  Object {
    "dateEnd": 1507888800000,
    "dateStart": 1504864800000,
    "id": 3016,
    "location": Object {
      "code": "US",
      "name": "Atlanta, Georgia, USA",
    },
    "name": "ELEAGUE CS:GO Premier 2017",
    "numberOfTeams": 16,
    "prizePool": "$1,000,000",
  },
  Object {
    "dateEnd": 1507370400000,
    "dateStart": 1507197600000,
    "id": 3101,
    "location": Object {
      "code": "DE",
      "name": "Husum, Germany",
    },
    "name": "NGC Masters 2017",
    "numberOfTeams": 8,
    "prizePool": "$17,585",
  },
  Object {
    "dateEnd": 1506852000000,
    "dateStart": 1506679200000,
    "id": 3189,
    "location": Object {
      "code": "CN",
      "name": "Kunshan, China",
    },
    "name": "China Top 2017 Kunshan",
    "numberOfTeams": 4,
    "prizePool": "$100,000",
  },
  Object {
    "dateEnd": 1506247200000,
    "dateStart": 1506160800000,
    "id": 3046,
    "location": Object {
      "code": "US",
      "name": "Burbank, California, USA",
    },
    "name": "ESEA Season 25 MDL Global Challenge",
    "numberOfTeams": 8,
    "prizePool": "$50,000",
  },
  Object {
    "dateEnd": 1505642400000,
    "dateStart": 1505469600000,
    "id": 2636,
    "location": Object {
      "code": "US",
      "name": "New York, New York, USA",
    },
    "name": "ESL One New York 2017",
    "numberOfTeams": 8,
    "prizePool": "$250,000",
  },
  Object {
    "dateEnd": 1505037600000,
    "dateStart": 1504864800000,
    "id": 2973,
    "location": Object {
      "code": "DK",
      "name": "Sønderborg, Denmark",
    },
    "name": "Cross Border Esport 2017",
    "numberOfTeams": 24,
    "prizePool": "$15,300",
  },
  Object {
    "dateEnd": 1505037600000,
    "dateStart": 1504864800000,
    "id": 2572,
    "location": Object {
      "code": "CA",
      "name": "Montreal, Canada",
    },
    "name": "DreamHack Open Montreal 2017",
    "numberOfTeams": 8,
    "prizePool": "$100,000",
  },
  Object {
    "dateEnd": 1505037600000,
    "dateStart": 1504778400000,
    "id": 3061,
    "location": Object {
      "code": "GR",
      "name": "Mykonos, Greece",
    },
    "name": "ESG Tour Mykonos 2017",
    "numberOfTeams": 8,
    "prizePool": "$235,000",
  },
  Object {
    "dateEnd": 1504432800000,
    "dateStart": 1504346400000,
    "id": 3140,
    "location": Object {
      "code": "MO",
      "name": "Ilhas, Macau",
    },
    "name": "GIRLGAMER Esports Festival 2017",
    "numberOfTeams": 4,
    "prizePool": "$12,781",
  },
  Object {
    "dateEnd": 1504432800000,
    "dateStart": 1504087200000,
    "id": 2684,
    "location": Object {
      "code": "SE",
      "name": "Malmö, Sweden",
    },
    "name": "DreamHack Masters Malmö 2017",
    "numberOfTeams": 16,
    "prizePool": "$250,000",
  },
  Object {
    "dateEnd": 1504346400000,
    "dateStart": 1504346400000,
    "id": 3085,
    "location": Object {
      "code": "DE",
      "name": "Peine, Germany",
    },
    "name": "iiyama CS:GO HomeMasters #3 Finals",
    "numberOfTeams": 2,
    "prizePool": "$3,569",
  },
  Object {
    "dateEnd": 1504346400000,
    "dateStart": 1499508000000,
    "id": 2979,
    "location": Object {
      "code": "GB",
      "name": "London, United Kingdom",
    },
    "name": "Gfinity Elite Series Season 1",
    "numberOfTeams": 8,
    "prizePool": "$110,000",
  },
  Object {
    "dateEnd": 1503828000000,
    "dateStart": 1503741600000,
    "id": 3093,
    "location": Object {
      "code": "RU",
      "name": "Moscow, Russia",
    },
    "name": "Binary Dragons Prestige LAN",
    "numberOfTeams": 8,
    "prizePool": "$10,000",
  },
  Object {
    "dateEnd": 1500804000000,
    "dateStart": 1500631200000,
    "id": 2571,
    "location": Object {
      "code": "US",
      "name": "Atlanta, Georgia, USA",
    },
    "name": "DreamHack Open Atlanta 2017",
    "numberOfTeams": 8,
    "prizePool": "$100,000",
  },
  Object {
    "dateEnd": 1500112800000,
    "dateStart": 1499940000000,
    "id": 2570,
    "location": Object {
      "code": "ES",
      "name": "Valencia, Spain",
    },
    "name": "DreamHack Open Valencia 2017",
    "numberOfTeams": 8,
    "prizePool": "$100,000",
  },
  Object {
    "dateEnd": 1499594400000,
    "dateStart": 1499162400000,
    "id": 2635,
    "location": Object {
      "code": "DE",
      "name": "Cologne, Germany",
    },
    "name": "ESL One Cologne 2017",
    "numberOfTeams": 16,
    "prizePool": "$250,000",
  },
  Object {
    "dateEnd": 1498989600000,
    "dateStart": 1498730400000,
    "id": 2721,
    "location": Object {
      "code": "RO",
      "name": "Bucharest, Romania",
    },
    "name": "PGL Major Krakow 2017 Main Qualifier",
    "numberOfTeams": 16,
    "prizePool": "Other",
  },
  Object {
    "dateEnd": 1498384800000,
    "dateStart": 1498212000000,
    "id": 2763,
    "location": Object {
      "code": "GB",
      "name": "London, United Kingdom",
    },
    "name": "ECS Season 3 Finals",
    "numberOfTeams": 8,
    "prizePool": "$660,000",
  },
  Object {
    "dateEnd": 1498298400000,
    "dateStart": 1498298400000,
    "id": 2890,
    "location": Object {
      "code": "DE",
      "name": "Hamburg, Germany",
    },
    "name": "Legend Series #2 Finals",
    "numberOfTeams": 4,
    "prizePool": "$3,360",
  },
  Object {
    "dateEnd": 1497866400000,
    "dateStart": 1497693600000,
    "id": 2924,
    "location": Object {
      "code": "SE",
      "name": "Jönköping, Sweden",
    },
    "name": "ESU Masters 2017",
    "numberOfTeams": 8,
    "prizePool": "$11,468",
  },
  Object {
    "dateEnd": 1497866400000,
    "dateStart": 1497693600000,
    "id": 2569,
    "location": Object {
      "code": "SE",
      "name": "Jönköping, Sweden",
    },
    "name": "DreamHack Open Summer 2017",
    "numberOfTeams": 8,
    "prizePool": "$100,000",
  },
  Object {
    "dateEnd": 1497780000000,
    "dateStart": 1497780000000,
    "id": 2889,
    "location": Object {
      "code": "RU",
      "name": "Moscow, Russia",
    },
    "name": "Adrenaline Cyber League 2017",
    "numberOfTeams": 4,
    "prizePool": "$100,000",
  },
  Object {
    "dateEnd": 1497780000000,
    "dateStart": 1497693600000,
    "id": 2884,
    "location": Object {
      "code": "AU",
      "name": "Sydney, Australia",
    },
    "name": "ZEN Esports Network League Season 1 Finals",
    "numberOfTeams": 4,
    "prizePool": "$37,184",
  },
  Object {
    "dateEnd": 1497780000000,
    "dateStart": 1497520800000,
    "id": 2723,
    "location": Object {
      "code": "RO",
      "name": "Bucharest, Romania",
    },
    "name": "Europe Minor - PGL Major Krakow 2017",
    "numberOfTeams": 8,
    "prizePool": "Other",
  },
  Object {
    "dateEnd": 1497607200000,
    "dateStart": 1497607200000,
    "id": 2853,
    "location": Object {
      "code": "US",
      "name": "Atlanta, Georgia, USA",
    },
    "name": "ELEAGUE Clash for Cash",
    "numberOfTeams": 2,
    "prizePool": "$250,000",
  },
  Object {
    "dateEnd": 1497175200000,
    "dateStart": 1496916000000,
    "id": 2724,
    "location": Object {
      "code": "RU",
      "name": "Moscow, Russia",
    },
    "name": "CIS Minor - PGL Major Krakow 2017",
    "numberOfTeams": 8,
    "prizePool": "$50,000",
  },
  Object {
    "dateEnd": 1497175200000,
    "dateStart": 1496916000000,
    "id": 2722,
    "location": Object {
      "code": "US",
      "name": "Santa Ana, California, USA",
    },
    "name": "Americas Minor - PGL Major Krakow 2017",
    "numberOfTeams": 8,
    "prizePool": "$50,000",
  },
  Object {
    "dateEnd": 1496570400000,
    "dateStart": 1496311200000,
    "id": 2725,
    "location": Object {
      "code": "CN",
      "name": "Beijing, China",
    },
    "name": "Asia Minor - PGL Major Krakow 2017",
    "numberOfTeams": 8,
    "prizePool": "$50,000",
  },
  Object {
    "dateEnd": 1496570400000,
    "dateStart": 1496138400000,
    "id": 2557,
    "location": Object {
      "code": "US",
      "name": "Dallas, Texas, USA",
    },
    "name": "ESL Pro League Season 5 Finals",
    "numberOfTeams": 12,
    "prizePool": "$750,000",
  },
  Object {
    "dateEnd": 1495965600000,
    "dateStart": 1495879200000,
    "id": 2777,
    "location": Object {
      "code": "PL",
      "name": "Wroclaw, Poland",
    },
    "name": "GeForce Cup 2017",
    "numberOfTeams": 8,
    "prizePool": "$30,000",
  },
  Object {
    "dateEnd": 1495879200000,
    "dateStart": 1495879200000,
    "id": 2834,
    "location": Object {
      "code": "RS",
      "name": "Belgrade, Serbia",
    },
    "name": "The Adria Masters Season 1 Finals",
    "numberOfTeams": 4,
    "prizePool": "$5,000",
  },
  Object {
    "dateEnd": 1494756000000,
    "dateStart": 1494669600000,
    "id": 2801,
    "location": Object {
      "code": "RO",
      "name": "Bucharest, Romania",
    },
    "name": "HTC 2v2 Invitational by PGL",
    "numberOfTeams": 8,
    "prizePool": "$50,000",
  },
  Object {
    "dateEnd": 1494756000000,
    "dateStart": 1494669600000,
    "id": 2749,
    "location": Object {
      "code": "GB",
      "name": "Leicester, United Kingdom",
    },
    "name": "ESEA Season 24 MDL Global Challenge",
    "numberOfTeams": 8,
    "prizePool": "$50,000",
  },
  Object {
    "dateEnd": 1494237600000,
    "dateStart": 1494064800000,
    "id": 2568,
    "location": Object {
      "code": "FR",
      "name": "Tours, France",
    },
    "name": "DreamHack Open Tours 2017",
    "numberOfTeams": 8,
    "prizePool": "$100,000",
  },
  Object {
    "dateEnd": 1494151200000,
    "dateStart": 1494064800000,
    "id": 2807,
    "location": Object {
      "code": "PT",
      "name": "Lisbon, Portugal",
    },
    "name": "4Gamers CS:GO Masters",
    "numberOfTeams": 6,
    "prizePool": "$10,997",
  },
  Object {
    "dateEnd": 1494151200000,
    "dateStart": 1493805600000,
    "id": 2713,
    "location": Object {
      "code": "AU",
      "name": "Sydney, Australia",
    },
    "name": "IEM Sydney 2017",
    "numberOfTeams": 8,
    "prizePool": "$200,000",
  },
  Object {
    "dateEnd": 1493546400000,
    "dateStart": 1493373600000,
    "id": 2567,
    "location": Object {
      "code": "US",
      "name": "Austin, Texas, USA",
    },
    "name": "DreamHack Open Austin 2017",
    "numberOfTeams": 8,
    "prizePool": "$100,000",
  },
  Object {
    "dateEnd": 1492941600000,
    "dateStart": 1492682400000,
    "id": 2641,
    "location": Object {
      "code": "US",
      "name": "Los Angeles, California",
    },
    "name": "cs_summit Spring 2017",
    "numberOfTeams": 8,
    "prizePool": "$150,000",
  },
  Object {
    "dateEnd": 1492250400000,
    "dateStart": 1492164000000,
    "id": 2753,
    "location": Object {
      "code": "DK",
      "name": "Copenhagen, Denmark",
    },
    "name": "Copenhagen Games 2017",
    "numberOfTeams": 16,
    "prizePool": "$56,300",
  },
  Object {
    "dateEnd": 1492250400000,
    "dateStart": 1492077600000,
    "id": 2663,
    "location": Object {
      "code": "DK",
      "name": "Copenhagen, Denmark",
    },
    "name": "Copenhagen Games 2017 Female",
    "numberOfTeams": 28,
    "prizePool": "$15,025",
  },
  Object {
    "dateEnd": 1492164000000,
    "dateStart": 1492077600000,
    "id": 2675,
    "location": Object {
      "code": "DK",
      "name": "Copenhagen, Denmark",
    },
    "name": "Copenhagen Games 2017 BYOC Qualifier",
    "numberOfTeams": 76,
    "prizePool": "Other",
  },
  Object {
    "dateEnd": 1491732000000,
    "dateStart": 1491300000000,
    "id": 2683,
    "location": Object {
      "code": "UA",
      "name": "Kiev, Ukraine",
    },
    "name": "SL i-League StarSeries Season 3 Finals",
    "numberOfTeams": 16,
    "prizePool": "$300,000",
  },
  Object {
    "dateEnd": 1491040800000,
    "dateStart": 1491040800000,
    "id": 2742,
    "location": Object {
      "code": "DE",
      "name": "Siek, Germany",
    },
    "name": "Legend Series #1 Finals",
    "numberOfTeams": 2,
    "prizePool": "$2,917",
  },
  Object {
    "dateEnd": 1489921200000,
    "dateStart": 1489748400000,
    "id": 2160,
    "location": Object {
      "code": "CN",
      "name": "Chengde, China",
    },
    "name": "China Cup 2017",
    "numberOfTeams": 6,
    "prizePool": "$99,000",
  },
  Object {
    "dateEnd": 1488711600000,
    "dateStart": 1488366000000,
    "id": 2637,
    "location": Object {
      "code": "PL",
      "name": "Katowice, Poland",
    },
    "name": "IEM Katowice 2017",
    "numberOfTeams": 12,
    "prizePool": "$250,000",
  },
  Object {
    "dateEnd": 1488538800000,
    "dateStart": 1488452400000,
    "id": 2662,
    "location": Object {
      "code": "PL",
      "name": "Katowice, Poland",
    },
    "name": "Intel Challenge Katowice 2017",
    "numberOfTeams": 8,
    "prizePool": "$30,000",
  },
  Object {
    "dateEnd": 1487502000000,
    "dateStart": 1487156400000,
    "id": 2541,
    "location": Object {
      "code": "US",
      "name": "Las Vegas, Nevada, USA",
    },
    "name": "DreamHack Masters Las Vegas 2017",
    "numberOfTeams": 16,
    "prizePool": "$450,000",
  },
  Object {
    "dateEnd": 1486206000000,
    "dateStart": 1486119600000,
    "id": 2649,
    "location": Object {
      "code": "FI",
      "name": "Helsinki, Finland",
    },
    "name": "Assembly Winter 2017",
    "numberOfTeams": 16,
    "prizePool": "$15,000",
  },
  Object {
    "dateEnd": 1484478000000,
    "dateStart": 1484305200000,
    "id": 2566,
    "location": Object {
      "code": "DE",
      "name": "Leipzig, Germany",
    },
    "name": "DreamHack ASTRO Open Leipzig 2017",
    "numberOfTeams": 8,
    "prizePool": "$100,000",
  },
  Object {
    "dateEnd": 1484478000000,
    "dateStart": 1484218800000,
    "id": 2564,
    "location": Object {
      "code": "CN",
      "name": "Changzhou, China",
    },
    "name": "WESG 2016 World Finals",
    "numberOfTeams": 24,
    "prizePool": "$1,500,000",
  },
  Object {
    "dateEnd": 1483873200000,
    "dateStart": 1483700400000,
    "id": 2538,
    "location": Object {
      "code": "US",
      "name": "Burbank, California, USA",
    },
    "name": "ESEA Season 23 MDL Global Challenge",
    "numberOfTeams": 8,
    "prizePool": "$50,000",
  },
  Object {
    "dateEnd": 1483873200000,
    "dateStart": 1483614000000,
    "id": 2642,
    "location": Object {
      "code": "SE",
      "name": "Stockholm, Sweden",
    },
    "name": "ASUS ROG Community Challenge Grand Finals 2017",
    "numberOfTeams": 8,
    "prizePool": "Other",
  },
]
`;
